src/ralph/backends/data/clickhouse.py

```diff
@@ -382,6 +382,23 @@ def write(  # pylint: disable=too-many-arguments
 
         return count
 
+    def close(self) -> None:
+        """Close the ClickHouse backend client.
+
+        Raise:
+            BackendException: If a failure occurs during the close operation.
+        """
+        if not self._client:
+            logger.warning("No backend client to close.")
+            return
+
+        try:
+            self.client.close()
```

Comment:
What about setting `self._client` to `None` afterward, to allow backends to create a new connection if requested?

Reply:
Not sure if it would be useful, as we would could call the `close` method of the backend either:
- when shutting down the FastAPI server,
- before garbage collecting the backend on a CLI command.
Either way, we would have to re-instantiate a backend.
Besides, it would create differences between backends, as `mongo` and `async_mongo` create their connections when being instantiated (no `_client` attribute mechanism)
WDYT?

Reply:
Good point) Indeed, the usefulness of this change might be quite limited, probably when used in a library context?
I guess we could address this later on if needed)